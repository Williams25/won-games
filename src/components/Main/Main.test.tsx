import { render, screen } from "@testing-library/react";
import { Main } from "./index";

describe("Main", () => {
  it("should render the heading", async () => {
    render(
      <Main title="React avançado" description="TypeScript, React, NextJS" />
    );
    expect(
      screen.getByRole("heading", { name: /react avançado/i })
    ).toBeInTheDocument();
  });

  it("should render correctly", async () => {
    const { container } = render(
      <Main title="React avançado" description="TypeScript, React, NextJS" />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
