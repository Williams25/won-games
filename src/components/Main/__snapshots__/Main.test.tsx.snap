// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render correctly 1`] = `
.c0 {
  background: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 3rem;
}

.c1 {
  width: 25rem;
  margin-bottom: 2rem;
}

.c2 {
  font-size: 2.5rem;
}

.c3 {
  font-size: 2rem;
  font-weight: 400;
}

.c4 {
  margin-top: 3rem;
  width: min(30rem,100%);
}

<main
  class="c0"
>
  <img
    alt="Imagem de um átomo e react avançado escrito ao lado"
    class="c1"
    src="/img/logo.svg"
  />
  <h1
    class="c2"
  >
    React avançado
  </h1>
  <h2
    class="c3"
  >
    TypeScript, React, NextJS
  </h2>
  <img
    alt="um desenvolver de frete ao computador"
    class="c4"
    src="/img/hero-illustration.svg"
  />
</main>
`;
